### DISCUSSION TITLE
Error handling in WEB UI

### DISCUSSION THREAD

**Discussion by z3dev** [jscad/OpenJSCAD.org#1066] [Ideas]
I'm having a very painful realization that the error handling inside the WEB UI is broken badly.

Basically, there is no error handling inside the worker thread, so thrown errors eventually get caught by the window and become an ErrorEvent. (At least that's my understanding currently). That little lack of oversight means that MOST is handling the error, which of course ends up a totally useless event that bubbles into the streams. Alas, everything is lost; no file name, no line numbers, no stack trace, etc.

I started down the road to make the error messages better for the users. And have had some success. But it feels like a hack now. The error handling has to be fixed.

Ideally, all errors should be caught inside the worker thread, handled nicely, and reported back to the main thread via a callback with type "error". However, making this happen means working within the MOST spaghetti code again.

I'm discussing this issue because @hrgdavor is working on the WEB UI. The error handling has to be thought out well so this does not happen again.

**Comment by hrgdavor** [comment]
Yes that is a must have, nad yes it is tricky because it also can vary between browsers, and when loading code into strigs to produse a function reported errors can also have line number shifted by some ammount.

The work is in two areas.
- script eval code in util (I think)
- worker code to report err
